### PR TITLE
Eigen, CLAPACK and libffi submodules

### PR DESCRIPTION
**Pull request by Shillaker**
Related to https://github.com/faasm/faasm/issues/314 

This includes moving the relevant Python tooling into this repo (from Faasm)

### CODE REVIEW

File: faasmtools/build.py [hunk 1]
```diff
@@ -0,0 +1,154 @@
+from copy import copy
+from os.path import join
+from subprocess import run
+
+from .env import PROJ_ROOT
+
```
Comment (Shillaker) [inline faasmtools/build.py:6]: This has been moved from Faasm

File: setup.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+setuptools.setup(
+    name="faasm-toolchain",
+    version=version,
+    author="Simon S",
+    author_email="blah@foo.com",
+    description="Utilities related to the Faasm toolchain",
+    url="https://github.com/faasm/faasm-toolchain",
+    python_requires=">=3.6",
+)
```
Comment (Shillaker) [inline setup.py:20]: Making this into a Python package lets us use the tools in this repo in other projects

File: tasks/eigen.py [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    THIRD_PARTY_DIR,
+)
+
+from faasmtools.build import (
+    CMAKE_TOOLCHAIN_FILE,
+    WASM_SYSROOT,
+)
+
+
+@task(default=True)
```
Comment (Shillaker) [inline tasks/eigen.py:19]: This was moved in from Faasm

File: tasks/libffi.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+from subprocess import run
+
+from faasmtools.build import (
+    build_config_cmd,
+    run_autotools,
+    WASM_SYSROOT,
+    WASM_LIB_INSTALL,
+)
+
+from invoke import task
```
Comment (Shillaker) [inline tasks/libffi.py:11]: This was moved in from faasm-cpython

**Review by csegarragonz — APPROVED** [review]

**Pull request merged**